### PR TITLE
add accessors to cache objects

### PR DESCRIPTION
**Pull request by abelcastilloavant**
Addresses https://github.com/kirillseva/cacher/issues/1

So yeah, overloading the `[[` operator is risky, but at least it's doable.

Edit: less risky than I originally thought.

### CODE REVIEW

**Comment by abelcastilloavant** [comment]
Well then - that broke a lot of tests.

File: R/accessors.R [hunk 1]
```diff
@@ -0,0 +1,4 @@
+`[[.LRUcache_` <- function(cache, key) { cache$get(key) }
+`[[<-.LRUcache_` <- function(cache, key, value) {
+  cache$set(key, value = value)
+}
```
Comment (kirillseva) [inline R/accessors.R]: don't you need to export them?

File: tests/testthat/test-accessors.R [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  x <- LRUcache(1)
+  x$set("a", 10)
+  expect_equal(x[["a"]], 10)
+})
+
+test_that("`[[` allows assignment of objects to cache", {
+  y <- LRUcache(1)
+  y[["b"]] <- 20
+  expect_equal(y$get("b"), 20)
+})
```
Comment (kirillseva) [inline tests/testthat/test-accessors.R:11]: It's great that tests work locally, but I'm pretty sure that those methods need to be `@export`'ed

**Review by kirillseva — CHANGES_REQUESTED** [review]
Please remove your vim swap files from the PR

**Review by kirillseva — APPROVED** [review]

**Pull request merged**